File: snobot_sim/cpp/lib/SnobotSim/MotorSim/GravityLoadDcMotorSim.cpp
/*
 * GravityLoadDcMotorSim.cpp
 *
 *  Created on: May 9, 2017
 *      Author: preiniger
 */

#include "SnobotSim/MotorSim/GravityLoadDcMotorSim.h"

const double GravityLoadDcMotorSim::sGRAVITY = 9.8;

GravityLoadDcMotorSim::GravityLoadDcMotorSim(const DcMotorModel& aMotorModel, double aLoad) :
        BaseDcMotorSimulator("Gravity Load", aMotorModel), mLoad(aLoad)
{

}

GravityLoadDcMotorSim::~GravityLoadDcMotorSim()
{

}


void GravityLoadDcMotorSim::Update(double cycleTime)
{
    double extraAcceleration = -sGRAVITY;

    mMotorModel.Step(mVoltagePercentage * 12, mLoad, extraAcceleration, cycleTime);
}

File: snobot_sim_java/src/main/java/com/snobot/simulator/motor_sim/GravityLoadDcMotorSim.java
package com.snobot.simulator.motor_sim;

public class GravityLoadDcMotorSim extends BaseDcMotorSimulator
{
    protected static final double sGRAVITY = 9.8;

    protected final GravityLoadMotorSimulationConfig mConfig;

    public GravityLoadDcMotorSim(DcMotorModel aModel, GravityLoadMotorSimulationConfig aConfig)
    {
        super(aModel);

        mConfig = aConfig;
    }


    @Override
    public void update(double aCycleTime)
    {
        double extraAcceleration = -sGRAVITY;

        mMotorModel.step(mVoltagePercentage * 12, mConfig.getLoad(), extraAcceleration, aCycleTime);
    }

    public GravityLoadMotorSimulationConfig getConfig()
    {
        return mConfig;
    }
}
